examples/livehunt_network_watch.py

```diff
@@ -0,0 +1,224 @@
+#!/usr/local/bin/python
+# Copyright © 2022 The vt-py authors. All Rights Reserved.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Manage a set of YARA nethunting rules by adding/removing domains.
+
+This script automatically updates a set of nethunting rules (file, url, domain,
+ip) in your VT account by adding and removing domains.
+
+Read more:
+https://www.virustotal.com/gui/hunting-overview
+https://developers.virustotal.com/v3.0/reference#livehunt
+https://developers.virustotal.com/docs/nethunt
+"""
+
+import argparse
+import asyncio
+import copy
+import os
+import re
+import vt
+import sys
+
+
+API_KEY_ENV_VAR = 'VT_API_KEY'
+RULESET_PREFIX = 'auto_network_watch_'
+
+
+def extract_domains_from_rule(rules):
+  """Extract the domain list from the comment of a yara rule."""
+  domain_list = []
+  for line in rules.split('*/')[0].split('---', 2)[1].splitlines():
+    if not line:
+      continue
+    domain = line.split('* ', 2)[1]
+    if domain:
+      domain_list.append(domain)
+  return domain_list
+
+
+def get_rulesets():
+  """Retrieve a rule from VT to get currently monitored properties."""
+  rulesets = {}
+  with vt.Client(os.environ.get(API_KEY_ENV_VAR)) as client:
+    try:
+      rulesets_it = client.iterator(
+        '/intelligence/hunting_rulesets',
+        params={'filter': f'name:{RULESET_PREFIX}* tag:autogenerated'},
+        limit=10)
+    except vt.error.APIError as e:
+      print(f'Error retrieving {RULESET_PREFIX}* rulesets: {e}')
+
+    if rulesets_it:
+      for ruleset in rulesets_it:
+        kind = ruleset.name.split(RULESET_PREFIX)[1]
+        rulesets[kind] = {'id': ruleset.id,
+                          'name': ruleset.name,
+                          'rules': ruleset.rules,
+                          'domains': extract_domains_from_rule(ruleset.rules)}
+    return rulesets
+
+
+def render_template(kind, domains):
+  template_dir = os.path.join(os.path.dirname(__file__), 'netwatch_templates')
+  body_template = os.path.join(template_dir, '_body.yara')
+  if not os.path.exists(body_template):
+    print(f'ERROR: file {body_template} not found.')
+    sys.exit(1)
```

Comment:
Maybe it's best to raise an exception here and handle the error on the main function?